Boost build don't seem to work with GCC on darwin arm64

### Make sure you completed the following tasks

- [x] I searched the [discussions](https://github.com/bfgroup/b2/discussions)
- [x] I searched the closed and open [issues](https://github.com/bfgroup/b2/issues?q=is%3Aissue)
- [x] I read the [contribution guidelines](https://github.com/bfgroup/b2/blob/main/CONTRIBUTING.adoc)

### Environment and version details

* Operating System+version: **MacOS 12.3.1 (M1 ARM)**
* Compiler+version: **GCC 11.2**
* Shell: **nix shell**
* B2 Version: B2 Version 4.7
* B2 Configuration:

```
notice: found boost-build.jam at /Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/boost-build.jam
notice: loading B2 from /Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/kernel/bootstrap.jam
notice: Searching '/etc' '/Users/myself' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/kernel' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/kernel' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/util' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/build' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/tools' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/contrib' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/.' for site-config configuration file 'site-config.jam'.
notice: Configuration file 'site-config.jam' not found in '/etc' '/Users/myself' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/kernel' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/kernel' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/util' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/build' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/tools' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/contrib' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/.'.
notice: Searching '/Users/myself' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/kernel' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/kernel' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/util' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/build' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/tools' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/contrib' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/.' for user-config configuration file 'user-config.jam'.
notice: Configuration file 'user-config.jam' not found in '/Users/myself' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/kernel' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/kernel' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/util' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/build' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/tools' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/contrib' '/Users/myself/project/build/vcpkg_installed/arm64-osx/tools/boost-build/src/.'.
notice: [cmdline-cfg] toolset gcc not previously configured; attempting to auto-configure now
notice: will use 'g++' for gcc, condition <toolset>gcc-11
notice: using gcc libraries :: <toolset>gcc-11 :: /nix/store/a74idnqpcb9wr9p30fza69wzvcir72pf-gcc-wrapper-11.2.0/bin /nix/store/a74idnqpcb9wr9p30fza69wzvcir72pf-gcc-wrapper-11.2.0/lib /nix/store/a74idnqpcb9wr9p30fza69wzvcir72pf-gcc-wrapper-11.2.0/lib32 /nix/store/a74idnqpcb9wr9p30fza69wzvcir72pf-gcc-wrapper-11.2.0/lib64
notice: using gcc archiver :: <toolset>gcc-11 :: ar
warning: toolset gcc initialization: can not find tool windres
warning: initialized from
notice: using rc compiler :: <toolset>gcc-11 :: /nix/store/a74idnqpcb9wr9p30fza69wzvcir72pf-gcc-wrapper-11.2.0/bin/as
...found 1 target...
```

### Brief problem description

Any boost build commands ends with `lol_add failed due to reached limit of 19 elements`, even the debug commands

### Steps to reproduce the issue

Create those files in a directory:
`vcpkg.json`
```js
{
    "name": "test-b2",
    "version-string": "0.1",
    "dependencies": [
        "boost-system"
    ]
}
```
`CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.21)
project(testb2 CXX)
add_executable(e main.cpp)
```
`main.cpp`
```cpp
#include <cstdio>
int main() { std::puts("hello gcc"); }
```
`vcpkg.nix`
```nix
{
  stdenv,
  fetchFromGitHub,
  gcc,
  ninja,
  cmake,
  lib,
  git,
  cacert,
  runtimeShell
}:

stdenv.mkDerivation rec {
  pname = "vcpkg";
  version = "74ff2a04415691361b07b7dbd1d4762ad79bd82c";

  src = fetchFromGitHub {
    owner = "microsoft";
    repo = "vcpkg-tool";
    rev = "2022-03-30";
    sha256 = "06l0zpqlh20dc300zlf47xjpk5x2db390pdir77jyk3cfpx7i3x2";
  };

  vcpkg_src = fetchFromGitHub {
    owner = "microsoft";
    repo = "vcpkg";
    rev = "74ff2a04415691361b07b7dbd1d4762ad79bd82c";
    sha256 = "0w32hz2055gjj33p2yi7h5sqn3ccv5c5wa6m9k7nxb56g3l05cjs";
  };

  buildInputs = [cmake ninja git cacert];

  vcpkgScript = ''
    #!${runtimeShell}
    vcpkg_hash=$(echo -n "${placeholder "out"}" | sha256sum | cut -f1 -d ' ')
    vcpkg_root_path="$HOME/.vcpkg/roots/$vcpkg_hash"
    if [[ ! -d "$vcpkg_root_path" ]]; then
      mkdir -p "$vcpkg_root_path"
      touch "$vcpkg_root_path/.vcpkg-root"
      ln -s ${placeholder "out"}/share/vcpkg/{docs,ports,scripts,triplets,versions} "$vcpkg_root_path/"
      ln -s ${placeholder "out"}/bin/vcpkg "$vcpkg_root_path/"
    fi
    VCPKG_DOWNLOADS="$vcpkg_root_path/downloads" VCPKG_ROOT="$vcpkg_root_path" ${placeholder "out"}/share/vcpkg/vcpkg "$@"
  '';
  
  passAsFile = [ "vcpkgScript" ];
  installPhase = ''
    cmake --build . --target=install --config=Release
    mkdir -p $out/share/vcpkg
    cp --preserve=mode -r ${vcpkg_src}/{docs,ports,scripts,triplets,.vcpkg-root,versions} $out/share/vcpkg
    mv $out/bin/vcpkg $out/share/vcpkg/vcpkg
    cp $vcpkgScriptPath $out/bin/vcpkg
    chmod +x $out/bin/vcpkg
    touch $out/share/vcpkg/vcpkg.disable-metrics
  '';
  enableParallelBuilding = true;
}
```
`shell.nix`
```nix
{
    pkgs   ? import <nixpkgs> {},
    stdenv ? pkgs.gcc11Stdenv
}:
rec {
  testb2 = stdenv.mkDerivation rec {
    name = "testb2";
    version = "dev-0.1";
    vcpkg = pkgs.callPackage ./vcpkg.nix {};
    buildInputs = with pkgs; [
      gcc11
      ninja
      cmake
      git
      perl
      gdb
      vcpkg

      pkg-config-unwrapped
      
      # vcpkg dependencies
      curl
      zip
      unzip
      cacert
    ];

    hardeningDisable = [ "stackprotector" ];
    
    VCPKG_ROOT = "~/.vcpkg/roots/${builtins.hashString "sha256" "${vcpkg}"}";
    shellHook = ''
      vcpkg > /dev/null 2>&1
    '';
  };
}
```
Run those commands in the terminal where the files were created:
```sh
nix-shell
mkdir build && cd build
cmake .. -DCMAKE_TOOLCHAIN_FILE=${VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake -DCMAKE_CXX_COMPILER=g++ -GNinja
```
Problem also reported at microsoft/vcpkg#24659

Boost build was compiled with `-fno-stack-protector` since it would not compile with stack protection.

### Actual behavior summary

The output of any `b2` command will output the following:
```
notice: found boost-build.jam at /Users/software/test2/build/vcpkg_installed/arm64-osx/tools/boost-build/boost-build.jam
notice: loading B2 from /Users/software/test2/build/vcpkg_installed/arm64-osx/tools/boost-build/src/kernel/bootstrap.jam
lol_add failed due to reached limit of 19 elements
```

**If a project of yours is blocked due to this bug, please, mention it explicitly.**

Our development may continue on Linux virtual machines instead

### Expected behavior summary

The error should not happen and `b2` should work as well as if it was compiled using clang.

It's just unbelievable, we got a Heisenbug here. I've found the problem could be made to appear and disappear in a totally unbelievable manner.

I was trying to replicate it with Homebrew instead of Gentoo prefix.

First, git checkout to branch 99742ad614537370686ab488560c1b23c6b4c95b, this is the official 4.9.3 version tag. Running `./bootstrap.sh gcc-12` creates a `b4` with a size of 342991 bytes. This executable will always crash. Next, git checkout to just one commit beyond 4.9.3, which is e9d5539b127361a8eab57d0892ca175435e49354 - this is just a merge commit without real changes, but running `./bootstrap.sh gcc-12` again produces a working `b4` with a size of 661151 bytes!

Now I suspect it's perhaps a complier bug that could be influenced by the filesystem. Alternatively, git history is not really linear, and perhaps the commit e9d5539b127361a8eab57d0892ca175435e49354 has something hidden?

## Broken Version

```
bash-5.2$ ./bootstrap.sh gcc-12
Building the B2 engine..

###
###
### Using 'gcc-12' toolset.
###
###

g++-12 (Homebrew GCC 12.2.0) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


###
###

> g++-12 -x c++ -std=c++11 -O2 -s -DNDEBUG builtins.cpp class.cpp command.cpp compile.cpp constants.cpp cwd.cpp debug.cpp debugger.cpp execcmd.cpp execnt.cpp execunix.cpp filesys.cpp filent.cpp fileunix.cpp frames.cpp function.cpp glob.cpp hash.cpp hcache.cpp hdrmacro.cpp headers.cpp jam_strings.cpp jam.cpp jamgram.cpp lists.cpp make.cpp make1.cpp md5.cpp mem.cpp modules.cpp native.cpp object.cpp option.cpp output.cpp parse.cpp pathnt.cpp pathsys.cpp pathunix.cpp regexp.cpp rules.cpp scan.cpp search.cpp startup.cpp subst.cpp sysinfo.cpp timestamp.cpp variable.cpp w32_getreg.cpp modules/order.cpp modules/path.cpp modules/property-set.cpp modules/regex.cpp modules/sequence.cpp modules/set.cpp -o b2
builtins.cpp: In function 'LIST* builtin_calc(FRAME*, int)':
builtins.cpp:530:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  530 |     sprintf( buffer, "%ld", result_value );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from builtins.cpp:7:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
builtins.cpp: In function 'LIST* builtin_backtrace(FRAME*, int)':
builtins.cpp:1381:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1381 |         sprintf( buf, "%d", line );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
builtins.cpp: In function 'LIST* builtin_nearest_user_location(FRAME*, int)':
builtins.cpp:1680:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1680 |         sprintf( buf, "%d", line );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
builtins.cpp: In function 'LIST* builtin_md5(FRAME*, int)':
builtins.cpp:1713:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1713 |         sprintf( hex_output + di * 2, "%02x", digest[ di ] );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
builtins.cpp: In function 'LIST* builtin_file_open(FRAME*, int)':
builtins.cpp:1754:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1754 |         sprintf( buffer, "%d", fd );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
builtins.cpp: In function 'LIST* builtin_shell(FRAME*, int)':
builtins.cpp:2120:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 2120 |         sprintf( buffer, "%d", exit_status );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
debugger.cpp: In function 'void debug_parent_clear(int, const char**)':
debugger.cpp:1523:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1523 |     sprintf( buf, "%d", id );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/wchar.h:90,
                 from /opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/cwchar:44,
                 from /opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/bits/postypes.h:40,
                 from /opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/bits/char_traits.h:39,
                 from /opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/string:40,
                 from object.h:16,
                 from debugger.h:13,
                 from debugger.cpp:8:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
debugger.cpp: In function 'void debug_parent_backtrace(int, const char**)':
debugger.cpp:1581:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1581 |         sprintf( buf, "%d", i );
      |         ~~~~~~~^~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
debugger.cpp: In function 'void debug_mi_break_insert(int, const char**)':
debugger.cpp:1941:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1941 |         sprintf( buf, "%d", num_breakpoints );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
execcmd.cpp: In function 'void argv_from_shell(const char**, LIST*, const char*, int32_t)':
execcmd.cpp:55:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
   55 |     sprintf( jobno, "%d", slot + 1 );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from execcmd.cpp:11:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
filesys.cpp: In function 'void file_archivescan_impl(OBJECT*, archive_scanback, void*)':
filesys.cpp:421:20: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  421 |             sprintf( buf, "%s(%s)",
      |             ~~~~~~~^~~~~~~~~~~~~~~~
  422 |                 object_str( archive->file->name ),
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  423 |                 object_str( member_file->name ) );
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from filesys.cpp:30:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
fileunix.cpp: In function 'void file_archscan(const char*, scanback, void*)':
fileunix.cpp:240:20: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  240 |             sprintf( buf, "%s(%s)",
      |             ~~~~~~~^~~~~~~~~~~~~~~~
  241 |                 object_str( archive->file->name ),
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  242 |                 object_str( member_file->name ) );
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from fileunix.cpp:30:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
fileunix.cpp: In function 'int file_collect_archive_content_(file_archive_info_t*)':
fileunix.cpp:357:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  357 |         sprintf( buf, "%s", lar_name );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
hcache.cpp: In function 'void hcache_done()':
hcache.cpp:397:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  397 |         sprintf( includes_count_str, "%lu", (long unsigned)list_length(
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  398 |             c->includes ) );
      |             ~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from hcache.cpp:32:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
hcache.cpp:399:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  399 |         sprintf( hdrscan_count_str, "%lu", (long unsigned)list_length(
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  400 |             c->hdrscan ) );
      |             ~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
hcache.cpp:401:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  401 |         sprintf( time_secs_str, "%lu", (long unsigned)c->time.secs );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
hcache.cpp:402:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  402 |         sprintf( time_nsecs_str, "%lu", (long unsigned)c->time.nsecs );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
hcache.cpp:403:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  403 |         sprintf( age_str, "%lu", (long unsigned)c->age );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
make.cpp: In function 'const char* target_name(TARGET*)':
make.cpp:776:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  776 |         sprintf( buf, "%s (internal node)", object_str( t->name ) );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from make.cpp:33:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
output.cpp: In function 'OBJECT* outf_int(int)':
output.cpp:176:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  176 |     sprintf( buffer, "%i", value );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from output.cpp:7:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
output.cpp: In function 'OBJECT* outf_double(double)':
output.cpp:184:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  184 |     sprintf( buffer, "%f", value );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
pathsys.cpp: In function 'OBJECT* path_tmpnam()':
pathsys.cpp:270:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  270 |     sprintf( name_buffer, "jam%lx%lx.000", pid, t );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from pathsys.cpp:30:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
scan.cpp: In function 'char* symdump(YYSYMBOL*)':
scan.cpp:716:29: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  716 |         case EOF   : sprintf( buf, "EOF"                                        ); break;
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from scan.cpp:12:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
scan.cpp:717:29: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  717 |         case 0     : sprintf( buf, "unknown symbol %s", object_str( s->string ) ); break;
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
scan.cpp:718:29: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  718 |         case ARG   : sprintf( buf, "argument %s"      , object_str( s->string ) ); break;
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
scan.cpp:719:29: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  719 |         case STRING: sprintf( buf, "string \"%s\""    , object_str( s->string ) ); break;
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
scan.cpp:720:29: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  720 |         default    : sprintf( buf, "keyword %s"       , s->keyword              ); break;
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
timestamp.cpp: In function 'const char* timestamp_formatstr(const timestamp*, const char*)':
timestamp.cpp:185:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  185 |     sprintf( result2, result1, time->nsecs );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from timestamp.cpp:26:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
ld: warning: option -s is obsolete and being ignored

Building is done. To install, run:

    ./b2 install --prefix=<DIR>

bash-5.2$ ./src/engine/b2
Segmentation fault: 11
bash-5.2$ ls -l ./src/engine/b2
-rwxr-xr-x  1 homebrew  staff  342991 Feb 20 00:54 ./src/engine/b2
```

## Good Version

```
bash-5.2$ git checkout e9d5539b127361a8eab57d0892ca175435e49354
Previous HEAD position was 99742ad61 Bump to 4.9.3.
HEAD is now at e9d5539b1 Merge branch 'version/4.9.3'

bash-5.2$ git clean -dxf ./
Removing b2
Removing src/engine/b2
bash-5.2$ ./bootstrap.sh gcc-12
Building the B2 engine..

###
###
### Using 'gcc-12' toolset.
###
###

g++-12 (Homebrew GCC 12.2.0) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


###
###

> g++-12 -x c++ -std=c++11 -O2 -s -DNDEBUG bindjam.cpp builtins.cpp class.cpp command.cpp compile.cpp constants.cpp cwd.cpp debug.cpp debugger.cpp execcmd.cpp execnt.cpp execunix.cpp filent.cpp filesys.cpp fileunix.cpp frames.cpp function.cpp glob.cpp hash.cpp hcache.cpp hdrmacro.cpp headers.cpp jam_strings.cpp jam.cpp jamgram.cpp lists.cpp make.cpp make1.cpp md5.cpp mem.cpp modules.cpp native.cpp option.cpp output.cpp parse.cpp pathnt.cpp pathsys.cpp pathunix.cpp regexp.cpp rules.cpp scan.cpp search.cpp startup.cpp subst.cpp timestamp.cpp value.cpp variable.cpp w32_getreg.cpp mod_jam_builtin.cpp mod_jam_class.cpp mod_jam_errors.cpp mod_jam_modules.cpp mod_order.cpp mod_path.cpp mod_property_set.cpp mod_regex.cpp mod_sequence.cpp mod_set.cpp mod_string.cpp mod_sysinfo.cpp mod_version.cpp -o b2
builtins.cpp: In function 'LIST* builtin_calc(FRAME*, int)':
builtins.cpp:521:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  521 |     sprintf( buffer, "%ld", result_value );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from builtins.cpp:7:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
builtins.cpp: In function 'LIST* builtin_backtrace(FRAME*, int)':
builtins.cpp:1372:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1372 |         sprintf( buf, "%d", line );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
builtins.cpp: In function 'LIST* builtin_nearest_user_location(FRAME*, int)':
builtins.cpp:1659:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1659 |         sprintf( buf, "%d", line );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
builtins.cpp: In function 'LIST* builtin_md5(FRAME*, int)':
builtins.cpp:1692:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1692 |         sprintf( hex_output + di * 2, "%02x", digest[ di ] );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
builtins.cpp: In function 'LIST* builtin_file_open(FRAME*, int)':
builtins.cpp:1733:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1733 |         sprintf( buffer, "%d", fd );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
builtins.cpp: In function 'LIST* builtin_shell(FRAME*, int)':
builtins.cpp:2075:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 2075 |         sprintf( buffer, "%d", exit_status );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
debugger.cpp: In function 'void debug_parent_clear(int, const char**)':
debugger.cpp:1523:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1523 |     sprintf( buf, "%d", id );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/wchar.h:90,
                 from /opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/cwchar:44,
                 from /opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/bits/postypes.h:40,
                 from /opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/bits/char_traits.h:39,
                 from /opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/string:40,
                 from value.h:15,
                 from object.h:12,
                 from debugger.h:13,
                 from debugger.cpp:8:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
debugger.cpp: In function 'void debug_parent_backtrace(int, const char**)':
debugger.cpp:1581:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1581 |         sprintf( buf, "%d", i );
      |         ~~~~~~~^~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
debugger.cpp: In function 'void debug_mi_break_insert(int, const char**)':
debugger.cpp:1941:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
 1941 |         sprintf( buf, "%d", num_breakpoints );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
execcmd.cpp: In function 'void argv_from_shell(const char**, LIST*, const char*, int32_t)':
execcmd.cpp:55:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
   55 |     sprintf( jobno, "%d", slot + 1 );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from execcmd.cpp:11:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
filesys.cpp: In function 'void file_archivescan_impl(OBJECT*, archive_scanback, void*)':
filesys.cpp:421:20: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  421 |             sprintf( buf, "%s(%s)",
      |             ~~~~~~~^~~~~~~~~~~~~~~~
  422 |                 object_str( archive->file->name ),
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  423 |                 object_str( member_file->name ) );
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from filesys.cpp:30:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
fileunix.cpp: In function 'void file_archscan(const char*, scanback, void*)':
fileunix.cpp:240:20: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  240 |             sprintf( buf, "%s(%s)",
      |             ~~~~~~~^~~~~~~~~~~~~~~~
  241 |                 object_str( archive->file->name ),
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  242 |                 object_str( member_file->name ) );
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from fileunix.cpp:30:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
fileunix.cpp: In function 'int file_collect_archive_content_(file_archive_info_t*)':
fileunix.cpp:357:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  357 |         sprintf( buf, "%s", lar_name );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
hcache.cpp: In function 'void hcache_done()':
hcache.cpp:397:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  397 |         sprintf( includes_count_str, "%lu", (long unsigned)list_length(
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  398 |             c->includes ) );
      |             ~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from hcache.cpp:32:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
hcache.cpp:399:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  399 |         sprintf( hdrscan_count_str, "%lu", (long unsigned)list_length(
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  400 |             c->hdrscan ) );
      |             ~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
hcache.cpp:401:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  401 |         sprintf( time_secs_str, "%lu", (long unsigned)c->time.secs );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
hcache.cpp:402:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  402 |         sprintf( time_nsecs_str, "%lu", (long unsigned)c->time.nsecs );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
hcache.cpp:403:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  403 |         sprintf( age_str, "%lu", (long unsigned)c->age );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
make.cpp: In function 'const char* target_name(TARGET*)':
make.cpp:776:16: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  776 |         sprintf( buf, "%s (internal node)", object_str( t->name ) );
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from make.cpp:33:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
output.cpp: In function 'OBJECT* outf_int(int)':
output.cpp:176:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  176 |     sprintf( buffer, "%i", value );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from output.cpp:7:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
output.cpp: In function 'OBJECT* outf_double(double)':
output.cpp:184:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  184 |     sprintf( buffer, "%f", value );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
pathsys.cpp: In function 'OBJECT* path_tmpnam()':
pathsys.cpp:270:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  270 |     sprintf( name_buffer, "jam%lx%lx.000", pid, t );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from pathsys.cpp:30:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
scan.cpp: In function 'char* symdump(YYSYMBOL*)':
scan.cpp:785:29: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  785 |         case EOF   : sprintf( buf, "EOF"                                        ); break;
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from scan.cpp:12:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
scan.cpp:786:29: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  786 |         case 0     : sprintf( buf, "unknown symbol %s", object_str( s->string ) ); break;
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
scan.cpp:787:29: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  787 |         case ARG   : sprintf( buf, "argument %s"      , object_str( s->string ) ); break;
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
scan.cpp:788:29: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  788 |         case STRING: sprintf( buf, "string \"%s\""    , object_str( s->string ) ); break;
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
scan.cpp:789:29: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  789 |         default    : sprintf( buf, "keyword %s"       , s->keyword              ); break;
      |                      ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
timestamp.cpp: In function 'const char* timestamp_formatstr(const timestamp*, const char*)':
timestamp.cpp:185:12: warning: 'int sprintf(char*, const char*, ...)' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Wdeprecated-declarations]
  185 |     sprintf( result2, result1, time->nsecs );
      |     ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from jam.h:348,
                 from timestamp.cpp:26:
/opt/homebrew/Cellar/gcc/12.2.0/lib/gcc/current/gcc/aarch64-apple-darwin22/12/include-fixed/stdio.h:204:10: note: declared here
  204 | int      sprintf(char * __restrict, const char * __restrict, ...) __printflike(2, 3);
      |          ^~~~~~~
ld: warning: option -s is obsolete and being ignored

Building is done. To install, run:

    ./b2 install --prefix=<DIR>

bash-5.2$ ./src/engine/b2
warning: No toolsets are configured.
warning: Configuring default toolset "clang".
warning: If the default is wrong, your build may not work correctly.
warning: Use the "toolset=xxxxx" option to override our guess.
warning: For more configuration options, please consult
warning: https://www.bfgroup.xyz/b2/manual/release/index.html#bbv2.overview.configuration
...found 1 target...

bash-5.2$ ls -l ./src/engine/b2
-rwxr-xr-x  1 homebrew  staff  661151 Feb 20 00:56 ./src/engine/b2
```

--- a/src/engine/modules/property-set.cpp
+++ b/src/engine/modules/property-set.cpp
@@ -162,7 +162,7 @@ LIST * property_set_create( FRAME * frame, int flags )
         OBJECT * rulename = object_new( "new" );
         OBJECT * varname = object_new( "self.raw" );
         LIST * val = call_rule( rulename, frame,
-            list_new( object_new( "property-set" ) ), 0 );
+            list_new( object_new( "property-set" ) ), (OBJECT *) 0 );
         LISTITER iter, end;
         object_free( rulename );
         pos->value = object_copy( list_front( val ) );
@@ -183,7 +183,7 @@ LIST * property_set_create( FRAME * frame, int flags )
                 import_module( imports, frame->module );
                 rulename = object_new( "errors.error" );
                 call_rule( rulename, frame,
-                    list_new( object_new( message->value ) ), 0 );
+                    list_new( object_new( message->value ) ), (OBJECT *) 0 );
                 /* unreachable */
                 string_free( message );
                 list_free( imports );

